### PR TITLE
Feature: zeb 599 test sessions ui

### PR DESCRIPTION
**Pull request by danymill**

### CODE REVIEW

File: client/app/modules/tests-sessions/test-session-card/test-session-card.scss [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+        &-left {
+            width: 40%;
+            flex-grow: 1;
+            min-width: 400px;
+        }
+        &-right {
+            width: 60%;
+            position: relative;
+        }
+        &.COMPLETED {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/test-session-card/test-session-card.scss]: I think it's better to use `lowercase` pipe in the html and use lowercase for class name

File: client/app/modules/tests-sessions/test-session-card/test-session-card.scss [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+            min-width: 200px;
+            padding-left: 11px;
+        }
+        &._env {
+            justify-content: center;
+            flex-grow: 1;
+            width: 11%;
+            min-width: 100px;
+            & .label {
+                font-weight: bold !important;
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/test-session-card/test-session-card.scss]: Ivan taught us to use numeric values for font-weight

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+    // }
+
+    function initAccessUrl() {
+        if (!vm.accessSettings) { return; }
+        if (vm.accessSettings.ZEBRUNNER_URL) {
+            try {
+                const url = new URL(vm.accessSettings.ZEBRUNNER_URL);
+
+                if (vm.accessSettings.ZEBRUNNER_USER) { url.username = vm.accessSettings.ZEBRUNNER_USER; }
+                if (vm.accessSettings.ZEBRUNNER_PASSWORD) {url.password = vm.accessSettings.ZEBRUNNER_PASSWORD; }
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: space

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+            return replacer && (typeof replacer === 'string' || typeof replacer === 'number') ? replacer : a;
+        });
+    }
+
+    function onCodeCopy() {
+        $timeout(() => {
+            vm.aceModel.copyToClipboard();
+            messageService.success('Code copied to clipboard');
+            $mdDialog.hide();
+        }, 0);
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: maybe it's better to use the third argument `false`

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                                    <input
+                                        type="text"
+                                        readonly
+                                        name="access-url"
+                                        ng-model="$ctrl.accessUrl"
+                                        ng-model-options="{debounce: 700}"/>
+                                </md-input-container>
+                            </div>
+                        </div>
+<!--                        <md-button ng-click="$ctrl.refreshAccessUrl()"-->
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: Add TODO

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+                </div>
+                <div class="access-key-modal__section _platforms"
+                    ng-class="{'_with-controls': $ctrl.platformControls.length}"
+                    ng-if="!$ctrl.isLoading && $ctrl.platforms.length">
+                    <div class="access-key-modal__platforms-header">
+                        <md-radio-group ng-model="$ctrl.platformModel[$ctrl.platformsConfig.rootKey]"
+                            layout="row"
+                            layout-wrap
+                            ng-change="$ctrl.onPlatformSelect()">
+<!--                            <md-radio-button ng-value="null">None</md-radio-button>-->
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: TODO:

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.scss [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    md-content {
+        background-color: transparent;
+    }
+
+    &__platforms {
+        &-header {
+            ._with-controls & {
+                margin-bottom: 15px;
+            }
+
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.scss]: space

File: client/app/modules/tests-sessions/test-session-card/test-session-card.scss [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+            color: rgba(0,0,0,.5);
+            margin-right: 4px;
+        }
+        &-text {
+            line-height: 14px;
+            margin-left: 2px;
+        }
+    }
+
+    @media screen and (max-width:  $breakpoint-mid-tablet) {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/test-session-card/test-session-card.scss]: space

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+        onModalApply,
+        platforms,
+    };
+
+    vm.$onInit = init;
+
+    return vm;
+
+    function init() {
+
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: add todo or remove it

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+            dateEnd: null,
+            showTemplate: false,
+            fullscreen: false
+        };
+        vm.searchParams = {};
+        onModalApply();
+    }
+
+    function openDatePicker($event, showTemplate) {
+
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: space

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        vm.selectedRange.showTemplate = showTemplate;
+
+        $mdDateRangePicker.show({
+            targetEvent: $event,
+            model: vm.selectedRange,
+            multiple: true,
+        })
+        .then(function(result) {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: incorrect tabs

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+            if (result) {
+                vm.selectedRange = result;
+                vm.selectedRange.selectedTemplateName = result.selectedTemplateName.split(' ').slice(0,-1).join(' ');
+                vm.searchParams.selectedTemplateName = vm.selectedRange.selectedTemplateName;
+                if (vm.selectedRange.dateStart && vm.selectedRange.dateEnd) {
+                    if (vm.selectedRange.dateStart.getTime() !==
+                        vm.selectedRange.dateEnd.getTime()) {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: bad codestyles

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+                } else {
+                    vm.searchParams.fromDate = null;
+                    vm.searchParams.toDate = null;
+                    vm.searchParams.date = null;
+                    vm.searchParams.selectedTemplateName = null;
+                }
+
+                vm.onChangeSearchCriteria();
+            }
+        })
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: add semicolon

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+                            <div class="zf-label" flex="none">Access URL</div>
+                            <div class="access-key-modal__control-input">
+                                <md-input-container class="no-margin _gray-bg">
+                                    <label>Access URL</label>
+                                    <input
+                                        type="text"
+                                        readonly
+                                        name="access-url"
+                                        ng-model="$ctrl.accessUrl"
+                                        ng-model-options="{debounce: 700}"/>
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: remove it

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+        platforms: [],
+        statuses: [],
+        scrollTicking: false,
+        selectedRange: {
+            selectedTemplate: null,
+            selectedTemplateName: null,
+            dateStart: null,
+            dateEnd: null,
+            showTemplate: false,
+            fullscreen: false
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: add comma

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+    function init() {
+        readDatePickerValues();
+        loadAdditionalSearchParams();
+        bindEventListeners();
+    }
+
+    function closeModal() {
+        $mdDialog.cancel();
+    }
+
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: remove space

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+    function showSearchDialog(event) {
+        $mdDialog.show({
+            controller: SearchModalController,
+            template: modalTemplate,
+            parent: angular.element(document.body),
+            targetEvent: event,
+            fullscreen: true,
+            controllerAs: '$ctrl',
+            onComplete: () => {
+                angular.element(window).on('resize.searchDialog',() => {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: add space

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 18]
```diff
@@ -0,0 +1,203 @@
...
+        vm.selectedRange.selectedTemplateName = null;
+        vm.searchParams = {};
+        onApply();
+    }
+
+    function onApply() {
+        $timeout(function() {
+            const params = {...testsSessionsService.DEFAULT_SC, ...vm.searchParams};
+
+            vm.onSearch({$params: params});
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: add spaces before/after brackets

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 19]
```diff
@@ -0,0 +1,203 @@
...
+        vm.selectedRange.dateEnd = null;
+        vm.selectedRange.selectedTemplate = null;
+        vm.selectedRange.selectedTemplateName = null;
+        vm.searchParams = {};
+        onApply();
+    }
+
+    function onApply() {
+        $timeout(function() {
+            const params = {...testsSessionsService.DEFAULT_SC, ...vm.searchParams};
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: add spaces before/after brackets

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 20]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    function openDatePicker($event, showTemplate) {
+        vm.selectedRange.showTemplate = showTemplate;
+
+        $mdDateRangePicker.show({
+            targetEvent: $event,
+            model: vm.selectedRange
+        })
+        .then(function(result) {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: tabs

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 21]
```diff
@@ -0,0 +1,203 @@
...
+            if (result) {
+                const res = UtilService.handleDateFilter(result);
+                const newSearchParams = {...vm.searchParams, ...res.searchParams};
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: add spaces before/after brackets

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 22]
```diff
@@ -0,0 +1,203 @@
...
+
+                vm.selectedRange = res.selectedRange;
+
+                if ((!res.searchParams.selectedTemplateName && vm.searchParams.selectedTemplateName) || (res.searchParams.selectedTemplateName && !angular.equals(vm.searchParams, newSearchParams))) {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: too long line

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 23]
```diff
@@ -0,0 +1,203 @@
...
+        });
+    }
+
+    function onScroll() {
+        if (!vm.scrollTicking) {
+            vm.scrollTicking = true;
+            $scope.$apply();
+            scrollTickingTimeout = $timeout(() => {
+                vm.scrollTicking = false;
+            }, 300);
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: magic number

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js [hunk 24]
```diff
@@ -0,0 +1,203 @@
...
+        } else {
+            $timeout.cancel(scrollTickingTimeout);
+            scrollTickingTimeout = $timeout(() => {
+                vm.scrollTicking = false;
+            }, 300);
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: magic number

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.html [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+                <span>{{$ctrl.selectedRange.selectedTemplateName || 'Date'}}</span>
+                <i class="material-icons">date_range</i>
+                <md-button id="searchCalendar" ng-click="$ctrl.openDatePicker($event, true)"></md-button>
+            </md-input-container>
+            <md-button ng-if="$ctrl.isSearchActive()" class="tests-sessions-search__btn md-raised pull-right" type="button" ng-click="$ctrl.onReset();"><span>RESET</span></md-button>
+        </div>
+    </div>
+    <md-fab-speed-dial id="main-fab" ng-if="$ctrl.isMobile && !$ctrl.scrollTicking" md-direction="up" class="md-scale md-fab-bottom-right">
+        <md-fab-trigger>
+            <md-button id="mainFab" aria-label="menu" class="md-fab md-primary" ng-click="$ctrl.showSearchDialog($event)" md-visible="tooltipVisible">
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.html]: add type for the button

File: client/app/modules/tests-sessions/tests-sessions.controller.js [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+        vm.testSessions = vm.resolvedTestSessions.results || [];
+        vm.totalResults = vm.resolvedTestSessions.totalResults || 0;
+        vm.pageSize = testsSessionsService.activeParams.pageSize;
+        vm.currentPage = testsSessionsService.activeParams.page + 1;
+        bindEvents();
+    }
+
+    // page value comes from pagination where counting starts from 1, so we need to subtract 1
+    function onPageChange(page) {
+        const activeParams = {...testsSessionsService.activeParams};
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions.controller.js]: spaces before/after brackets

File: client/app/modules/tests-sessions/tests-sessions.controller.js [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+
+        activeParams.page = page - 1;
+
+        return getTestSessions(activeParams);
+    }
+
+    function onSearch(params = {...testsSessionsService.DEFAULT_SC}) {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions.controller.js]: spaces before/after brackets

File: client/app/modules/tests-sessions/tests-sessions.controller.js [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+        vm.currentPage = params.page + 1;
+
+        return getTestSessions(params);
+    }
+
+    function getTestSessions(params = {...testsSessionsService.activeParams}) {
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions.controller.js]: spaces before/after brackets

File: client/app/modules/tests-sessions/tests-sessions.controller.js [hunk 29]
```diff
@@ -0,0 +1,104 @@
...
+        return testsSessionsService.searchSessions(params)
+            .then((rs) => {
+                if (rs.success) {
+                    const data = rs.data || {};
+
+                    vm.testSessions = data.results || [];
+                    vm.totalResults = data.totalResults || 0;
+
+                    return $q.resolve(vm.testSessions);
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions.controller.js]: you can use the data without $q

File: client/app/modules/tests-sessions/tests-sessions.controller.js [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+            .catch(function(err) {
+                messageService.error(err.message);
+
+                return $q.resolve([]);
+            });
+    }
+
+    function bindEvents() {
+        const onTransStartSubscription = $transitions.onStart({}, function(trans) {
+            // const toState = trans.to();
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions.controller.js]: TODO:

File: client/app/modules/tests-sessions/tests-sessions.service.js [hunk 31]
```diff
@@ -0,0 +1,43 @@
...
+    $httpParamSerializer,
+    UtilService
+) {
+    'ngInject';
+
+    const DEFAULT_SC = {
+        page: 0,
+        pageSize: 20,
+    };
+    let lastParams = {...DEFAULT_SC};
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions.service.js]: spaces before/after brackets

File: client/app/modules/tests-sessions/tests-sessions.service.js [hunk 32]
```diff
@@ -0,0 +1,43 @@
...
+    };
+
+    function searchSessions(params = lastParams) {
+        service.activeParams = params;
+
+        return $httpMock.get(`${API_URL}/api/tests/sessions/search?${$httpParamSerializer(service.activeParams)}`).then(UtilService.handleSuccess, UtilService.handleError('Unable to search test sessions'));
+    }
+
+    function resetCachedParams() {
+        lastParams = {...DEFAULT_SC};
```
Comment (DmitryMishin) [inline client/app/modules/tests-sessions/tests-sessions.service.js]: spaces before/after brackets

**Review by dzmitrymishyn — CHANGES_REQUESTED** [review]

File: client/app/_services/json-configs/json-configs.service.js [hunk 33]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+import languagesConfig from './languages-config.json';
+
+const jsonConfigsService = function jsonConfigsService(
+    $httpMock,
+    UtilService,
+
```
[the remark this answers is not in the document]
Comment (danymill) [reply client/app/_services/json-configs/json-configs.service.js]: fixed

File: client/app/_services/json-configs/json-configs.service.js [hunk 34]
```diff
@@ -0,0 +1,32 @@
...
+    $q
```
[the remark this answers is not in the document]
Comment (danymill) [reply client/app/_services/json-configs/json-configs.service.js]: fixed

File: client/app/modules/tests-sessions/test-session-card/test-session-card.scss (hunk 1 above)
Comment (danymill) [reply client/app/modules/tests-sessions/test-session-card/test-session-card.scss]: fixed

File: client/app/modules/tests-sessions/test-session-card/test-session-card.scss (hunk 2 above)
Comment (danymill) [reply client/app/modules/tests-sessions/test-session-card/test-session-card.scss]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js (hunk 3 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js (hunk 4 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html (hunk 5 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: fixed

File: client/app/modules/tests-sessions/tests-sessions.controller.js (hunk 28 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.controller.js (hunk 29 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.controller.js (hunk 30 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.service.js (hunk 31 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.service.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.service.js (hunk 32 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.service.js]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html (hunk 6 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.scss (hunk 7 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.scss]: fixed

File: client/app/modules/tests-sessions/test-session-card/test-session-card.scss (hunk 8 above)
Comment (danymill) [reply client/app/modules/tests-sessions/test-session-card/test-session-card.scss]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 9 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 10 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 11 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 12 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 13 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html (hunk 14 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 15 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html (hunk 6 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.scss (hunk 7 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.scss]: fixed

File: client/app/modules/tests-sessions/test-session-card/test-session-card.scss (hunk 8 above)
Comment (danymill) [reply client/app/modules/tests-sessions/test-session-card/test-session-card.scss]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 9 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 10 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 11 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 12 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js (hunk 13 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/modal/search-modal.controller.js]: fixed

File: client/app/modules/tests-sessions/access-key-modal/access-key-modal.html (hunk 14 above)
Comment (danymill) [reply client/app/modules/tests-sessions/access-key-modal/access-key-modal.html]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 15 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 16 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 17 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 19 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 20 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 21 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 22 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 23 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js (hunk 24 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.html (hunk 25 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions-search/tests-sessions-search.html]: fixed

File: client/app/modules/tests-sessions/tests-sessions.controller.js (hunk 27 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.controller.js (hunk 28 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.controller.js (hunk 29 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.controller.js (hunk 30 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.controller.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.service.js (hunk 31 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.service.js]: fixed

File: client/app/modules/tests-sessions/tests-sessions.service.js (hunk 32 above)
Comment (danymill) [reply client/app/modules/tests-sessions/tests-sessions.service.js]: fixed

**Review by dzmitrymishyn — APPROVED** [review]

**Pull request merged**